src/oatpp/network/server/Server.cpp

```diff
@@ -86,5 +105,18 @@ v_int32 Server::getStatus(){
   return m_status.load();
 }
 
+Server::~Server() {
+  switch (getStatus()) {
+    case STATUS_STARTING:
+    case STATUS_RUNNING:
+      stop();
+      break;
+    default:
+      // Just check if the thread is still alive and wait for it to exit
+      if (!m_blocking && m_thread.joinable()) {
```

Comment:
Let's keep all `stop()` relates checks in the stop method.
Let's have the destructor to be as follows:

```cpp
Server::~Server() {
  stop(); // wait here if server was not stopped
}
```